packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -903,112 +958,116 @@ export async function makeWallet({
     });
   }
 
+  /** @type {Store<Payment, PaymentRecord>} */
   const payments = makeStore('payment');
   const {
     updater: paymentsUpdater,
     notifier: paymentsNotifier,
-  } = makeNotifierKit([]);
+  } = /** @type {NotifierRecord<PaymentRecord[]>} */ (makeNotifierKit([]));
+  /**
+   * @param {PaymentRecord} param0
+   */
   const updatePaymentRecord = ({ actions, ...preDisplay }) => {
     const displayPayment = fillInSlots(dehydrate(harden(preDisplay)));
     const paymentRecord = { ...preDisplay, actions, displayPayment };
     payments.set(paymentRecord.payment, harden(paymentRecord));
     paymentsUpdater.updateState([...payments.values()]);
   };
 
+  /**
+   * @param {Payment} payment
+   * @param {Purse | Petname=} depositTo
+   */
   const addPayment = async (payment, depositTo = undefined) => {
-    /**
-     * @typedef {Object} PaymentRecord
-     * @property {Issuer} issuer
-     * @property {Payment} payment
-     * @property {Brand} brand
-     * @property {'pending'|'deposited'} status
-     * @property {typeof actions} actions
-     */
-
     // We don't even create the record until we get an alleged brand.
     const brand = await E(payment).getAllegedBrand();
 
-    /** @type {Partial<PaymentRecord>} */
-    let paymentRecord = {
+    /** @type {PaymentRecord} */
+    let paymentRecord;
+
+    const depositedPK = makePromiseKit();
+    paymentRecord = {
       payment,
       brand,
       issuer: undefined,
       status: undefined,
-    };
-
-    const depositedPK = makePromiseKit();
-    const actions = {
-      async deposit(purseOrPetname = undefined) {
-        let purse;
-        if (purseOrPetname === undefined) {
-          if (!brandToAutoDepositPurse.has(brand)) {
-            // No automatic purse right now.
-            return depositedPK.promise;
+      actions: {
+        async deposit(purseOrPetname = undefined) {
+          /** @type {Purse} */
+          let purse;
+          if (purseOrPetname === undefined) {
+            if (!brandToAutoDepositPurse.has(brand)) {
+              // No automatic purse right now.
+              return depositedPK.promise;
+            }
+            // Plop into the current autodeposit purse.
+            purse = brandToAutoDepositPurse.get(brand);
+          } else if (
+            Array.isArray(purseOrPetname) ||
```

Comment:
If it is an array here, what does that mean? Is it a path?

Reply:
Yes, it is a path.